include/SugarEmailAddress/SugarEmailAddress.php

```diff
@@ -2312,15 +2320,27 @@ public function getOptInStatusTickHTML()
                         break;
                 }
 
-                $template->assign('optInFlagClass', $optInFlagClass);
-                $template->assign('optInFlagTitle', $optInFlagTitle);
-                $template->assign('optInFlagText', $optInFlagText);
-                $tickHtml = $template->fetch('include/SugarEmailAddress/templates/optInStatusTick.tpl');
-            }
+            $template->assign('optInFlagClass', $optInFlagClass);
+            $template->assign('optInFlagTitle', $optInFlagTitle);
+            $template->assign('optInFlagText', $optInFlagText);
+            $tickHtml = $template->fetch('include/SugarEmailAddress/templates/optInStatusTick.tpl');
+        }
 
         return $tickHtml;
     }
-
+    
+    /**
+     *
+     * @return string
+     */
+    public function getConfirmOptInTokenGenerateIfNotExists()
+    {
+        if (!$this->confirm_opt_in_token) {
+            $this->confirm_opt_in_token = md5(time()) . md5($this->email_address) . md5(rand(0, 9999999)) . md5(rand(0, 9999999));
```

Comment:
I still have a rainbow table for this lol.

Reply:
good luck then..! :rainbow: :D